src/Messaging/SAF.Messaging.Redis/ServiceCollectionExtensions.cs

```diff
@@ -56,6 +56,25 @@ internal static IServiceCollection AddRedisMessagingInfrastructure(this IService
             return serviceCollection;
         }
 
+        /// <summary>
+        /// Try to connect several times and throw an error message if it fails.
+        /// </summary>
+        private static  IConnectionMultiplexer CreateRedisConnectionMultipleTimes(RedisConfiguration config, ILogger logger)
+        {
+            IConnectionMultiplexer con = CreateRedisConnection(config);
+            int testCount = 10;
+            while (!con.IsConnected)
```

Comment:
I remember of some events defined in the `IConnectionMultiplexer` interface named `ConnectionFailed` or `ConnectionRestored`. Is there maybe a way to listen on that events instead of "looping around" here?